proxy/src/bind.rs

```diff
@@ -292,7 +307,16 @@ where
         fn(S::Service) -> NormalizeUri<S::Service>
     >;
     fn new_service(&self) -> Self::Future {
-        self.inner.new_service().map(NormalizeUri::new)
+        let s = self.inner.new_service();
+        // This weird dance is so that the closure doesn't have to
+        // capture `self` and can just be a `fn` (so the `Map`)
+        // can be returned unboxed.
+        if self.was_absolute_form {
+            s.map(|inner| NormalizeUri::new(inner, true))
+        } else {
+            s.map(|inner| NormalizeUri::new(inner, false))
+        }
```

Comment:
huh? can't this just be:
```rust
let was_absolute = self.was_absolute_form;
s.map(|inner| NormalizeUri::new(inner, was_asbolute))
```

Reply:
Yeah, it could be that. i'm dumb.

Reply:
nevermind, that doesn't compile, because now it closes over an external variable again. i _knew_ I did that for a reason